url73e3057d61_empeeoh_BACnet4J_tgz-pJ8-com_serotonin_bacnet4j_test_OverflowTestJ8/src/com/serotonin/bacnet4j/service/confirmed/ConfirmedRequestService.java

```diff
@@ -0,0 +1,177 @@
+            return;
+        if (type == AtomicWriteFileRequest.TYPE_ID && services.isAtomicWriteFile()) // 7
+            return;
+        if (type == AddListElementRequest.TYPE_ID && services.isAddListElement()) // 8
+            return;
+        if (type == RemoveListElementRequest.TYPE_ID && services.isRemoveListElement()) // 9
+            return;
+        if (type == CreateObjectRequest.TYPE_ID && services.isCreateObject()) // 10
+            return;
+        if (type == DeleteObjectRequest.TYPE_ID && services.isDeleteObject()) // 11
+            return;
+        if (type == ReadPropertyRequest.TYPE_ID && services.isReadProperty()) // 12
+            return;
+        if (type == ReadPropertyConditionalRequest.TYPE_ID && services.isReadPropertyConditional()) // 13
+            return;
+        if (type == ReadPropertyMultipleRequest.TYPE_ID && services.isReadPropertyMultiple()) // 14
+            return;
+        if (type == WritePropertyRequest.TYPE_ID && services.isWriteProperty()) // 15
+            return;
+        if (type == WritePropertyMultipleRequest.TYPE_ID && services.isWritePropertyMultiple()) // 16
+            return;
+        if (type == DeviceCommunicationControlRequest.TYPE_ID && services.isDeviceCommunicationControl()) // 17
+            return;
+        if (type == ConfirmedPrivateTransferRequest.TYPE_ID && services.isConfirmedPrivateTransfer()) // 18
+            return;
+        if (type == ConfirmedTextMessageRequest.TYPE_ID && services.isConfirmedTextMessage()) // 19
+            return;
+        if (type == ReinitializeDeviceRequest.TYPE_ID && services.isReinitializeDevice()) // 20
+            return;
+        if (type == VtOpenRequest.TYPE_ID && services.isVtOpen()) // 21
+            return;
+        if (type == VtCloseRequest.TYPE_ID && services.isVtClose()) // 22
+            return;
+        if (type == VtDataRequest.TYPE_ID && services.isVtData()) // 23
+            return;
+        if (type == AuthenticateRequest.TYPE_ID && services.isAuthenticate()) // 24
+            return;
+        if (type == RequestKeyRequest.TYPE_ID && services.isRequestKey()) // 25
+            return;
+        if (type == ReadRangeRequest.TYPE_ID && services.isReadRange()) // 26
+            return;
+        if (type == LifeSafetyOperationRequest.TYPE_ID && services.isLifeSafetyOperation()) // 27
+            return;
+        if (type == SubscribeCOVPropertyRequest.TYPE_ID && services.isSubscribeCovProperty()) // 28
+            return;
+        if (type == GetEventInformation.TYPE_ID && services.isGetEventInformation()) // 29
+            return;
+
+        throw new BACnetErrorException(ErrorClass.device, ErrorCode.serviceRequestDenied);
+    }
+
+    public static ConfirmedRequestService createConfirmedRequestService(byte type, ByteQueue queue)
+            throws BACnetException {
+
+        if (type == AcknowledgeAlarmRequest.TYPE_ID) // 0
+            return new AcknowledgeAlarmRequest(queue);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem:**
 This line of code lacks validation when processing input data through the following parameter: &#39;queue&#39; (index: 1 | type: ByteQueue). The parameter is exposed to external callers, because its enclosing class and method are publicly accessible. This means that upstream validation, if it exists, can be bypassed. Other validated parameters: &#39;type&#39;. The same parameter type is validated here for example: url73e3057d61_empeeoh_BACnet4J_tgz-pJ8-com_serotonin_bacnet4j_test_OverflowTestJ8/src/com/serotonin/bacnet4j/apdu/Abort.java:82. Malicious, malformed, or unbounded inputs can cause unexpected runtime behavior or crashes, and can slow performance.

**Fix:**
 Add checks to ensure the validity of the parameter&#39;s value, such as testing it for nullness (for example, using the @NonNull annotation as [described in the Lombok library](https://projectlombok.org/features/NonNull)), emptiness, or equality. Or to prevent direct calls to it, [reduce the method&#39;s visibility](https://docs.oracle.com/javase/tutorial/java/javaOO/accesscontrol.html).

 Learn more about potential threats and guidance from the [Common Weakness Enumeration website](https://cwe.mitre.org/data/definitions/693.html) and the [OWASP Cheat Sheet series](https://cheatsheetseries.owasp.org/cheatsheets/Input_Validation_Cheat_Sheet.html).